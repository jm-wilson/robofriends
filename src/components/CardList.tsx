import React from 'react';
import Card from './Card';
import { IRobot } from '../containers/App';

interface ICardListProps {
  robots: IRobot[];
}

const CardList: React.FC<ICardListProps> = ({ robots }): JSX.Element => {
  return !robots.length ? (
    <div></div>
  ) : (
    <div>
      {robots.map((user, i) => {
        return <Card key={i} id={robots[i].id} name={robots[i].name} email={robots[i].email} />;
      })}
    </div>
  );
};

export default CardList;
