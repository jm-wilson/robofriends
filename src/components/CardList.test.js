import { shallow } from 'enzyme';
import React from 'react';
import CardList from './CardList';

it('should render CardList component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Snow',
      email: 'john@gmail.com',
    },
  ];
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});

it('should render empty CardList component', () => {
  expect(shallow(<CardList robots={[]} />)).toMatchSnapshot();
});
