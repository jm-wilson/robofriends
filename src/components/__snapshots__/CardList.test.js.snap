// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CardList component 1`] = `
<div>
  <Card
    email="john@gmail.com"
    id={1}
    key="0"
    name="John Snow"
  />
</div>
`;

exports[`should render an empty CardList component 1`] = `<div />`;
